src/core/tsi/transport_security.h

```diff
@@ -111,6 +111,9 @@ tsi_result tsi_construct_allocated_string_peer_property(
 tsi_result tsi_construct_string_peer_property_from_cstring(
     const char *name, const char *value, tsi_peer_property *property);
 
+/* Destroy plugin registration funcion. */
+void tsi_register_destroy_plugin(void (*destroy)(void));
```

Comment:
I understand that.  There will be some things that need to be initialized preemptively at initialization time and others that need to be initialized lazily, and that's fine.  If a particular implementation does not have anything that needs to be initialized preemptively, then the preemptive initialization can be a no-op.  But we should still offer a preemptive initialization hook for the cases where it is needed.

In this case, I suspect that you will need both preemptive and lazy initialization.  As per my comment above, I think you will need to initialize the mutex preemptively, and you can then use that mutex to initialize the channel and CQ lazily.